src/black/__init__.py

```diff
@@ -670,7 +692,17 @@ def reformat_one(
     """
     try:
         changed = Changed.NO
-        if not src.is_file() and str(src) == "-":
+        is_stdin = False
+
+        if str(src) == "-":
+            is_stdin = True
+        elif str(src).startswith(STDIN_PLACEHOLDER):
+            is_stdin = True
+            # Use the original name again in case we want to print something
+            # to the user
+            src = Path(str(src)[len(STDIN_PLACEHOLDER) :])
+
+        if not src.is_file() and is_stdin:
```

Comment:
Be careful, if a file actually exists at the custom `stdin-filename` path, Black will switch to using `format_file_in_place` since `not src.is_file()` would return `False`.

```console
(black) ichard26@acer-ubuntu:~/programming/oss/black$ echo "this_is_a_different_piece_of = 'code'" > temp.py
(black) ichard26@acer-ubuntu:~/programming/oss/black$ echo "print('egg spam')" | black --diff --stdin-filename="temp.py" -
--- temp.py	2020-10-23 20:35:39.055756 +0000
+++ temp.py	2020-10-23 20:36:24.060408 +0000
@@ -1 +1 @@
-this_is_a_different_piece_of = 'code'
+this_is_a_different_piece_of = "code"
would reformat temp.py
All done! ✨ 🍰 ✨
1 file would be reformatted.
```